es-app/src/CollectionSystemManager.cpp

```diff
@@ -369,149 +422,65 @@ std::string CollectionSystemManager::getValidNewCollectionName(std::string inNam
-	std::vector<std::string> systemsInUse = getSystemsFromConfig();
-	// get folders assigned to custom collections
-	std::vector<std::string> autoSys = getCollectionThemeFolders(false);
-	// get folder assigned to custom collections
-	std::vector<std::string> customSys = getCollectionThemeFolders(true);
-	// get folders assigned to user collections
-	std::vector<std::string> userSys = getUserCollectionThemeFolders();
-	// add them all to the list of systems in use
-	systemsInUse.insert(systemsInUse.end(), autoSys.begin(), autoSys.end());
-	systemsInUse.insert(systemsInUse.end(), customSys.begin(), customSys.end());
-	systemsInUse.insert(systemsInUse.end(), userSys.begin(), userSys.end());
-	for(auto sysIt = systemsInUse.begin(); sysIt != systemsInUse.end(); sysIt++)
+
+	for(auto system : getSystemsInUse())
 	{
-		if (*sysIt == name)
+		if (system == name)
 		{
 			if(index > 0) {
 				name = name.substr(0, name.size()-4);
 			}
 			return getValidNewCollectionName(name, index+1);
 		}
 	}
-	// if it matches one of the custom collections reserved names
+	// if it matches one of the collections reserved names
 	if (mCollectionSystemDeclsIndex.find(name) != mCollectionSystemDeclsIndex.end())
 		return getValidNewCollectionName(name, index+1);
 	return name;
 }
 
 void CollectionSystemManager::setEditMode(std::string collectionName)
 {
-	if (mCustomCollectionSystemsData.find(collectionName) == mCustomCollectionSystemsData.end())
+
+	if (mCollectionSystems.find(collectionName) == mCollectionSystems.end() ||
+		!(mCollectionSystems.at(collectionName)->isEnabled))
 	{
-		LOG(LogError) << "Tried to edit a non-existing collection: " << collectionName;
+		LOG(LogError) << "Tried to edit a non-existing or disabled collection: " << collectionName;
+		mIsEditing = false;
 		return;
 	}
-	mIsEditingCustom = true;
-	mEditingCollection = collectionName;
 
-	CollectionSystemData* sysData = &(mCustomCollectionSystemsData.at(mEditingCollection));
-	if (!sysData->isPopulated)
+	mEditingCollection = collectionName;
+	mEditingCollectionSystem = mCollectionSystems.at(mEditingCollection);
+	if (!mEditingCollectionSystem->isPopulated)
 	{
-		populateCustomCollection(sysData);
+		mEditingCollectionSystem->populate();
 	}
-	// if it's bundled, this needs to be the bundle system
-	mEditingCollectionSystemData = sysData;
 
 	GuiInfoPopup* s = new GuiInfoPopup(mWindow, "Editing the '" + strToUpper(collectionName) + "' Collection. Add/remove games with Y.", 10000);
 	mWindow->setInfoPopup(s);
+	mIsEditing = true;
 }
 
 void CollectionSystemManager::exitEditMode()
 {
 	GuiInfoPopup* s = new GuiInfoPopup(mWindow, "Finished editing the '" + mEditingCollection + "' Collection.", 4000);
 	mWindow->setInfoPopup(s);
-	mIsEditingCustom = false;
-	mEditingCollection = "Favorites";
+	setEditMode(defaultEditingCollection);  //default edit mode.
+	
+		
+	//mEditingCollection = "Favorites";
+	// NOTE: this is a functional change,we no longer fall back on editing favorites by default.
```

Comment:
Huh. Tell me more. :)

Reply:
yeah, so this is related to the change you commented on in CollectionMAnager line 268, if Favs are no longer the only Auto-system, I thought we could just disable any editing, and have users get into favorites editing mode the same way as for any other collection (via the select-menu when in that particular collection view).

Reply:
Ah, got it. Makes perfect sense.